src/Sulu/Bundle/SecurityBundle/Resources/js/containers/Permissions/tests/Permissions.test.js

```diff
@@ -0,0 +1,821 @@
+                permissions: {
+                    'view': true,
+                    'delete': true,
+                    'add': true,
+                    'edit': true,
+                },
+            },
+            {
+                id: 2,
+                context: 'sulu.contact.organizations',
+                permissions: {
+                    'view': true,
+                    'delete': true,
+                    'add': true,
+                    'edit': false,
+                },
+            },
+        ];
+        permissions.find(PermissionMatrix).at(0).instance().props.onChange(newContextPermissions);
+        expect(onChange).toBeCalledWith(newContextPermissions);
+    });
+});
+
+test('Render with empty webspace section', () => {
+    const value: Array<ContextPermission> = [
+        {
+            id: 1,
+            context: 'sulu.contact.people',
+            permissions: {
+                'view': true,
+                'delete': true,
+                'add': true,
+                'edit': true,
+            },
+        },
+        {
+            id: 2,
+            context: 'sulu.contact.organizations',
+            permissions: {
+                'view': true,
+                'delete': true,
+                'add': true,
+                'edit': true,
+            },
+        },
+    ];
+
+    const securityContextGroups: SecurityContextGroups = {
+        'Contacts': {
+            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
+            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
+        },
+        'Webspaces': {
+            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
+            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
+            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
+        },
+    };
+    const promise = Promise.resolve(securityContextGroups);
+    securityContextsStore.loadSecurityContextGroups.mockReturnValue(promise);
+
+    // $FlowFixMe
+    ResourceListStore.mockImplementation(function() {
+        this.loading = false;
+        this.data = [
+            {
+                'key': 'example',
+                'name': 'Example',
+            },
+            {
+                'key': 'example2',
+                'name': 'Example 2',
+            },
+            {
+                'key': 'example3',
+                'name': 'Example 3!',
+            },
+        ];
+    });
+
+    const permissions = mount(
+        <Permissions
+            onChange={jest.fn()}
+            system={'Sulu'}
+            value={value}
+        />
+    );
+
+    return promise.then(() => {
+        expect(securityContextsStore.loadSecurityContextGroups).toBeCalledWith('Sulu');
+        permissions.update();
+
+        // Currently we have to load each child separately, because of a bug in enzyme.
+        // Fixme: https://github.com/airbnb/enzyme/issues/1213
```

Comment:
Would stay with `TODO`s, as we've always done until now, e.g. in https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/tests/Datagrid.test.js#L301